src/infra/src/table/migration/m20241115_150000_populate_folder_table.rs

```diff
@@ -0,0 +1,126 @@
+        // records into memory.
+        // txn.execute()
+        let mut meta_pages = meta::Entity::find()
+            .filter(meta::Column::Module.eq("folders"))
+            .order_by_asc(meta::Column::Id)
+            .paginate(&txn, 100);
+
+        while let Some(metas) = meta_pages.fetch_and_next().await? {
+            let folders_rslt: Result<Vec<_>, DbErr> = metas
+                .into_iter()
+                .map(|m| {
+                    // Transform unstructured JSON from the meta record.
+                    let json: MetaFolder =
+                        json::from_str(&m.value).map_err(|e| DbErr::Migration(e.to_string()))?;
+                    let description = if json.description.is_empty() {
+                        None
+                    } else {
+                        Some(json.description)
+                    };
+                    Ok(folder::ActiveModel {
+                        folder_id: Set(m.key2),
+                        org: Set(m.key1),
+                        name: Set(json.name),
+                        description: Set(description),
+                        r#type: Set(0),
+                        ..Default::default()
+                    })
+                })
+                .collect();
+            let folders = folders_rslt?;
+            folder::Entity::insert_many(folders).exec(&txn).await?;
+        }
+
+        txn.commit().await?;
+        Ok(())
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        let db = manager.get_connection();
+        folder::Entity::delete_many().exec(db).await?;
+        Ok(())
+    }
+}
+
+/// Representation of a folder in the meta table.
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MetaFolder {
+    #[serde(default)]
+    pub folder_id: String,
+    pub name: String,
+    pub description: String,
+}
+
+// The schemas of tables might change after subsequent migrations. Therefore
+// this migration only references ORM models in private submodules that should
+// remain unchanged rather than ORM models in the `entity` module that will be
+// updated to reflect the latest changes to table schemas.
+
+/// Representation of the meta table at the time this migration executes.
+mod meta {
+    use sea_orm::entity::prelude::*;
+
+    #[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
+    #[sea_orm(table_name = "meta")]
+    pub struct Model {
+        #[sea_orm(primary_key)]
+        pub id: i64,
+        pub module: String,
+        pub key1: String,
+        pub key2: String,
+        pub start_dt: i64,
+        #[sea_orm(column_type = "Text")]
+        pub value: String,
+    }
+
+    #[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
+    pub enum Relation {}
+
+    impl ActiveModelBehavior for ActiveModel {}
+}
+
+/// Representation of the folder table at the time this migration executes.
+mod folder {
```

Comment:
This module is an exact copy of the `folder` module in `src/infra/table/entity/folder.rs`. I think it's important that we copy those models here rather than referencing `src/infra/table/entity/folder.rs` for the following reasons:

The model here should to mirror the schema of the `folder` table at the exact point in history before this (and all subsequent migrations) run. Therefore it should never change, as changing this model could cause the migration logic to break.

In contrast, the model in `src/infra/table/entity/folder.rs` should represent the **latest** schema of the `folder`. It should change whenever we decide to change the schema of `folder` and the Sea ORM's automatic model generation script _will_ automatically change the model.